File: Leetcode/1. Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int> &numbers, int target) {
      
        int i,j,k,n;
        n=numbers.size();
        vector<pair<int,int> >p3a(n);
        for(i=0;i<n;i++)
        {
            p3a[i].first=numbers[i];
            p3a[i].second=i;
        }
        sort(p3a.begin(),p3a.end());
        i=0;j=n-1;
        vector<int>ans;
        while(i<j)
        {
            if(p3a[i].first+p3a[j].first==target)
            {
                ans.push_back(p3a[i].second+1);
                ans.push_back(p3a[j].second+1);
                sort(ans.begin(),ans.end());
                return ans;
            }
            else if(p3a[i].first+p3a[j].first>target)
            j--;
            else
            i++;
            
        }
        
    }
};

File: Leetcode/1. Two Sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int n = nums.length;
        List<Pair<Integer,Integer>> vals = new ArrayList<Pair<Integer,Integer>>();
        for(int i=0;i<n;i++){
            vals.add(new Pair(nums[i],i));
        }
        Collections.sort(vals, (v1,v2)-> v1.getKey().compareTo(v2.getKey()));
        int left = 0;
        int right = n-1;
        while(left<right){
            int currSum = vals.get(left).getKey() + vals.get(right).getKey();
            if(currSum == target){
                return new int[] {vals.get(left).getValue(),vals.get(right).getValue()};
            }
            else if(currSum < target){
                left++;
            }
            else{
                right--;
            }
        }
        return new int[]{};
    }
}
